src/encoded/batch_download.py

```diff
@@ -216,13 +247,123 @@ def peak_metadata(context, request):
+    for file in file_graph:
+        annotation = _get_embedded_schema_by_dataset(annotation_graph, file['dataset'])
+        software = _get_embedded_schema_by_dataset(software_graph, file['dataset'])
+        writer.writerow([
+            file.get('title', ''),
+            file.get('file_type', ''),
+            file.get('output_type', ''),
+            annotation.get('accession', ''),
+            annotation.get('annotation_type', ''),
+            software.get('title', ''),
+            annotation.get('encyclopedia_version', ''),
+            annotation.get('biosample_term_id', ''),
+            annotation.get('biosample_term_name', ''),
+            annotation.get('biosample_type', ''),
+            annotation.get('relevant_life_stage', ''),
+            annotation.get('relevant_timepoint', ''),
+            annotation.get('relevant_timepoint_units', ''),
+            annotation.get('organism', {}).get('scientific_name', ''),
+            annotation.get('targets', {}).get('name', ''),
+            annotation.get('date_released', ''),
+            annotation.get('award').get('project', ''),
+            file.get('derived_from', ''),
+            file.get('file_size', ''),
+            file.get('lab', {}).get('title', ''),
+            file.get('md5sum', ''),
+            file.get('dbxrefs', ''),
+            file.get('href', ''),
+            file.get('assembly', ''),
+            file.get('controlled_by', ''),
+            file.get('status', ''),
+            file.get('restricted', ''),
+        ])
+    return Response(
+        content_type='text/tsv',
+        body=fout.getvalue(),
+        content_disposition='attachment;filename="%s"' % 'metadata.tsv'
+    )
+
+
+def _get_embedded_schema_by_dataset(graph, dataset):
+    """
+    Get schema based on dataset.
+
+        :param graph: @graph value.
+        :param dataset: dataset.
+    """
+    schema = list(filter(lambda g: g.get('@id') == dataset, graph)) or [{}]
+    return schema[0]
+
+
+def _get_graph(schema_type, request, ids):
+    """
+    Get a graph based on schema type and id.
+
+        :param schema_type: schema type.
+        :param request: request object from pyramid.
+        :param ids: ids.
+    """
+    datasets = ''.join([''.join(['&dataset=', id]) for id in ids])
+    path = '/search/?type={}&status!=revoked{}&format=json'.format(
+        schema_type,
+        datasets,
+    )
+    results = request.embed(path, as_user=True)
+    graph = results['@graph']
+    return graph
+
 @view_config(route_name='metadata', request_method='GET')
 def metadata_tsv(context, request):
     param_list = parse_qs(request.matchdict['search_params'])
+    type_param = ('type' in param_list and param_list['type'][0] and param_list['type'][0]) or None
     if 'referrer' in param_list:
         search_path = '/{}/'.format(param_list.pop('referrer')[0])
     else:
         search_path = '/search/'
+    is_annotation = type_param and type_param.lower() == 'annotation'
```

Comment:
What happens if type not in ('annotation', 'experiment')???

Reply:
Great observation. You get a HTTP 400 response.

Reply:
I assume this does not need any further action.